src/pybind/mgr/cephadm/module.py

```diff
@@ -1582,9 +1582,27 @@ def _client_keyring_rm(
         self._kick_serve_loop()
         return HandleCommandResult()
 
-    def _get_container_image(self, daemon_name: str) -> Optional[str]:
+    def _get_container_image(self, daemon_name: str, use_current_daemon_image: bool = False) -> Optional[str]:
```

Comment:
The failure from run-tox-mgr looks like it has a problem with this new parameter:

```
Expected: _run_cephadm('hostx', 'smb.tango.briskly', ['_orch', 'deploy'], [], stdin='{"fsid": "fsid", "name": "smb.tango.briskly", "image": "", "deploy_arguments": [], "params": {}, "meta": {"service_name": "smb", "ports": [], "ip": null, "deployed_by": [], "rank": null, "rank_generation": null, "extra_container_args": null, "extra_entrypoint_args": null}, "config_blobs": {"cluster_id": "foxtrot", "features": ["domain"], "config_uri": "rados://.smb/foxtrot/config2.json", "join_sources": ["rados://.smb/foxtrot/join1.json", "rados:mon-config-key:smb/config/foxtrot/join2.json"], "custom_dns": ["10.8.88.103"], "config": "", "keyring": "[client.smb.config.tango.briskly]\\nkey = None\\n\\n[client.smb.fs.cephfs.share1]\\nkey = None\\n\\n[client.smb.fs.cephfs.share2]\\nkey = None\\n\\n[client.smb.fs.fs2.share3]\\nkey = None\\n", "config_auth_entity": "client.smb.config.tango.briskly"}}')
E           Actual: _run_cephadm('hostx', 'smb.tango.briskly', ['_orch', 'deploy'], [], stdin='{"fsid": "fsid", "name": "smb.tango.briskly", "image": "", "deploy_arguments": [], "params": {}, "meta": {"service_name": "smb", "ports": [], "ip": null, "deployed_by": [], "rank": null, "rank_generation": null, "extra_container_args": null, "extra_entrypoint_args": null}, "config_blobs": {"cluster_id": "foxtrot", "features": ["domain"], "config_uri": "rados://.smb/foxtrot/config2.json", "join_sources": ["rados://.smb/foxtrot/join1.json", "rados:mon-config-key:smb/config/foxtrot/join2.json"], "custom_dns": ["10.8.88.103"], "config": "", "keyring": "[client.smb.config.tango.briskly]\\nkey = None\\n\\n[client.smb.fs.cephfs.share1]\\nkey = None\\n\\n[client.smb.fs.cephfs.share2]\\nkey = None\\n\\n[client.smb.fs.fs2.share3]\\nkey = None\\n", "config_auth_entity": "client.smb.config.tango.briskly"}}', use_current_daemon_image=False)

/usr/lib/python3.10/unittest/mock.py:929: AssertionError
```

Reply:
Expected vs. actual is the same except for:

```
use_current_daemon_image=False
```

Reply:
I think I must have written this patch before rebasing on top of the recent SMB support patches in cephadm. Hopefully good now.